cmake/mirrors/custom.cmake

```diff
@@ -0,0 +1,70 @@
+# Gitlab-style mirror
+# CMake scripts look for opencv/opencv_3rdparty,
+#  OAID/Tengine, 01org/tbb(oneAPI/oneTBB), opencv/ade
+#  from OPENCV_DOWNLOAD_MIRROR
+ocv_update(OPENCV_DOWNLOAD_MIRROR_URL "")
+# Tengine
+ocv_update(TENGINE_CUSTOM_PKG_MD5 "")
+# TBB
+ocv_update(TBB_CUSTOM_RELEASE "")
+ocv_update(TBB_CUSTOM_PKG_NAME "")
+ocv_update(TBB_CUSTOM_PKG_MD5 "")
+# ADE
+ocv_update(ADE_CUSTOM_RELEASE "")
+ocv_update(ADE_CUSTOM_PKG_NAME "")
+ocv_update(ADE_CUSTOM_PKG_MD5 "")
+
+macro(ocv_download_url_custom_usercontent)
+  string(REPLACE "/" ";" DL_URL_split ${DL_URL})
+  list(GET DL_URL_split 5 __COMMIT_ID)
+  list(GET DL_URL_split 6 __PKG_NAME)
+  set(DL_URL "https://${OPENCV_DOWNLOAD_MIRROR_URL}/opencv/opencv_3rdparty/-/raw/${__COMMIT_ID}/${__PKG_NAME}/")
+endmacro()
+macro(ocv_download_url_custom_archive_commit_id)
+  if("m${${DL_ID}_CUSTOM_PKG_MD5}" STREQUAL "m")
+    message("ocv_download: set ${DL_ID}_CUSTOM_PKG_MD5 to download ${DL_ID} from custom source.")
```

Comment:
> set ...

This message looks misleading as we don't update downloaded MD5 hashsum.

Reply:
Yes, we do not update hashsum but users need to set values for several empty vars in custom.cmake to use their custom source. I think I can improve the description to tell users to modify the file.

Reply:
set -> specify ?

Reply:
Done.